src/java.base/share/classes/java/util/regex/Matcher.java

```diff
@@ -265,6 +265,10 @@ public Pattern pattern() {
         return parentPattern;
     }
 
+    private String captureText() {
+        return text instanceof String s ? s : text.subSequence(first, last).toString();
+    }
+
```

Comment:
I'd think we'd want to get a subSequence of any text, regardless of whether it's a String or some other CharSequence. This could result in a savings if there is a match result (or several match results) from find() operation(s) on a large String, when the matched text is relative small.

I'd also merge this into the creation of the MatchResult. The offset needs to be stored because the start/end locations are indexes into the original text, but we're storing only a subSequence of the original text. However, if we always store a subSequence, the offset is always the same as `first`.

Reply:
Done.